pkg/model/resource/gvk.go

```diff
@@ -60,8 +55,9 @@ func (gvk GVK) Validate() error {
 	if gvk.Version == "" {
 		return fmt.Errorf(versionRequired)
 	}
-	if !versionRegex.MatchString(gvk.Version) {
-		return fmt.Errorf("Version must match %s (was %s)", versionPattern, gvk.Version)
+	errs := validation.IsDNS1123Subdomain(gvk.Version)
+	if len(errs) > 0 && gvk.Version != versionInternal {
+		return fmt.Errorf("Version must respect DNS-1123 (was %s)", gvk.Version)
```

Comment:
Can we also wrap the errors returned and expose them, so that the user is aware on what is exactly wrong?